### PR TITLE
`Development`: Fix DOM event name conflicts

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
When merging develop the linter fails

> Cannot check the working tree for unmerged files because of an error. [STARTED] Preparing lint-staged... [COMPLETED] Preparing lint-staged... [STARTED] Running tasks for staged files... [STARTED] .lintstagedrc.js — 35 files [STARTED] src/{main/webapp,test}/**/*.{json,js,ts,css,scss,html} — 25 files [STARTED] src/{main/webapp,test}/**/*.{ts,html} — 24 files [STARTED] src/{main,test}/java/**/*.java — 4 files [STARTED] prettier --write [STARTED] eslint --fix [STARTED] bash ./linting.sh [COMPLETED] prettier --write [COMPLETED] src/{main/webapp,test}/**/*.{json,js,ts,css,scss,html} — 25 files [COMPLETED] bash ./linting.sh [COMPLETED] src/{main,test}/java/**/*.java — 4 files [FAILED] eslint --fix [FAILED] [FAILED] eslint --fix [FAILED] [COMPLETED] Running tasks for staged files... [STARTED] Applying modifications from tasks... [SKIPPED] Skipped because of errors from tasks. [STARTED] Reverting to original state because of errors... [COMPLETED] Reverting to original state because of errors... [STARTED] Cleaning up temporary files... [COMPLETED] Cleaning up temporary files... ✖ eslint --fix: /Users/florianglombik/IdeaProjects/Artemis-2024-08-05/src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.html 0:0 warning File ignored because no matching configuration was supplied /Users/florianglombik/IdeaProjects/Artemis-2024-08-05/src/main/webapp/app/exercises/programming/shared/code-editor/header/code-editor-header.component.html 0:0 warning File ignored because no matching configuration was supplied /Users/florianglombik/IdeaProjects/Artemis-2024-08-05/src/main/webapp/app/exercises/programming/shared/code-editor/monaco/code-editor-monaco.component.html 0:0 warning File ignored because no matching configuration was supplied /Users/florianglombik/IdeaProjects/Artemis-2024-08-05/src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts 24:15 error Output bindings, including aliases, should not be named as standard DOM events @angular-eslint/no-output-native ✖ 4 problems (1 error, 3 warnings) husky - pre-commit script failed (code 1)

Re-adding a lintignore that seems to be removed by accident in https://github.com/ls1intum/Artemis/commit/5e5e208dce052fd5e46ff9ff2cb28c9d8aa0462f

The issue appears to be that eslint in some cases simply removes 
`// eslint-disable-next-line @angular-eslint/no-output-native`

This does not appear to be the case for 

> // eslint-disable @angular-eslint/no-output-native -- Necessary to avoid conflicts with native DOM events
> @Output() cancel = new EventEmitter<void>(); // variable with conflict with native DOM element
> // eslint-enable @angular-eslint/no-output-native

As eslint optimizes away the disable next line and the disable/enable is rather cumbersome to deal with, I think it is a better approach to make sure this issue will not happen again by fixing the underlying naming convention issues.

### Description
II renamed the variables to resolve the conflict with the DOM events and conform (for my understanding) with angular conventions.

This is the case for the cancel and submit buttons in the assessment views of the exercises, where name conflicts existed.

### Steps for Testing
- 1 instructor

1. Go to an exercise at least one unassessed submission, start the assessment and verify you can cancel it
2. Go to an already assessed submission for an exercise an verify that overriding the assessment works
3. Verify assessment can be cancelled / submitted for modeling exercises
4. Verify assessment can be cancelled / submitted for file upload exercises
5. Verify assessment can be cancelled / submitted for text exercises
6. Verify assessment can be cancelled / submitted for programming exercises (you need to enable manual assessment in the exercise timeline)
7. Go to the programming exercise edit page and verify the cancel button works

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

## Summary by CodeRabbit

- **New Features**
	- Enhanced reactivity in the Form Footer component with new reactive properties for better context handling.
	- Renamed event emitters across various components for clarity, including `submit` to `onSubmit` and `cancel` to `onCancel`.

- **Bug Fixes**
	- Improved adherence to linting standards in the Form Footer component.

- **Refactor**
	- Adjusted event handling methods to align with new naming conventions across multiple components.
	- Updated logic for determining the save title in the Form Footer component based on reactive properties.

### CODE REVIEW

File: src/main/webapp/app/forms/form-footer/form-footer.component.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { ButtonSize } from 'app/shared/components/button.component';
 })
 export class FormFooterComponent {
     @Output() save = new EventEmitter<void>();
-    // eslint-disable-next-line @angular-eslint/no-output-native
+
     @Output() cancel = new EventEmitter<void>();
```
Comment (JohannesStoehr) [inline src/main/webapp/app/forms/form-footer/form-footer.component.ts:14]: The client style now complains here. Are you sure that we can remove the comment?

Comment (florian-glombik) [reply src/main/webapp/app/forms/form-footer/form-footer.component.ts:14]: Nah, I am trying to revert it right now, not sure which automatic save action removed it 😅

Comment (florian-glombik) [reply src/main/webapp/app/forms/form-footer/form-footer.component.ts:14]: eslint removed that line, thats probably also how the other one disappeared

Comment (florian-glombik) [reply src/main/webapp/app/forms/form-footer/form-footer.component.ts:14]: I have no changed the approach by directly changing the names, so the lintignore is not needed anymore and should not cause further issues in the future

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by krusche — APPROVED** [review]
The code looks good to me 👍 
I still suggest to test the features to make sure no rename was forgotten

**Review by HawKhiem — COMMENTED** [review]
I tested on TS1. Canceling assessment, overiding assessment works for all types of exercises. I can also cancel the editing of an exercise as well. However I did find some issues, which might not be directly linked to this PR, but I think should be resolved.

1. The overiding of assessment still seems a bit unintuitive for me, it only works if I directly edit my current assessment and that there is only 1 assessment, when I created a few new assessments and click on overide, it always takes the result of the best assessment (the one which gives the hightest points).

https://github.com/user-attachments/assets/12625039-36c2-4753-9676-758a42902330

2. I tried to assess a Upload exercise where the participant hasn't submitted anything. I presume this behaviour is currently not implemented
 

https://github.com/user-attachments/assets/078ddc98-2457-4bcf-a047-984519d7f2a7

3. I don't know if I did something wrong but somehow I couldn't upload a file in the upload exercise


https://github.com/user-attachments/assets/a31d4830-928c-4c9e-bfa1-a11feb40c009

4. Is it intended that 00000001 is equal to 1 in the assessment  section?

Screenshot 2024-10-26 171834

**Review by SimonEntholzer — APPROVED** [review]
Code changes make sense

**Pull request merged**